core/src/test/java/org/ehcache/EhcacheBasicPutIfAbsentEventsTest.java

```diff
@@ -0,0 +1,134 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.ehcache;
+
+import org.ehcache.event.CacheEvent;
+import org.ehcache.event.CacheEventListener;
+import org.ehcache.event.EventType;
+import org.ehcache.events.CacheEventNotificationService;
+import org.ehcache.events.CacheEventNotificationServiceImpl;
+import org.ehcache.exceptions.CacheAccessException;
+import org.ehcache.exceptions.CacheWritingException;
+import org.ehcache.spi.loaderwriter.CacheLoaderWriter;
+import org.ehcache.statistics.CacheOperationOutcomes;
+import org.ehcache.util.EhcacheBasicTestUtil;
+import org.ehcache.util.IsCreated;
+import org.ehcache.util.IsUpdated;
+import org.hamcrest.CoreMatchers;
+import org.junit.Test;
+import org.mockito.InOrder;
+import org.mockito.Mock;
+import org.slf4j.LoggerFactory;
+
+import java.util.Collections;
+import java.util.EnumSet;
+import java.util.Set;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.Matchers.nullValue;
+import static org.junit.Assert.assertThat;
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.fail;
+import static org.mockito.Matchers.any;
+import static org.mockito.Matchers.argThat;
+import static org.mockito.Matchers.eq;
+import static org.mockito.Mockito.doThrow;
+import static org.mockito.Mockito.inOrder;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.spy;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.verifyZeroInteractions;
+
+/**
+ * Provides testing of basic PUT_IF_ABSENT operations on an {@code Ehcache}.
+ *
+ * @author Clifford W. Johnson
+ */
+public class EhcacheBasicPutIfAbsentEventsTest extends EhcacheBasicCrudBase {
+
+  @Mock
+  protected CacheLoaderWriter<String, String> cacheLoaderWriter;
+
+  @Mock
+  protected CacheEventListener testCacheEventListener;
+
+  protected CacheEventNotificationService cacheEventNotificationService;
+
+  protected IsCreated isCreated = new IsCreated();
+
+  @Test
+  public void testPutIfAbsentNullNull() {
+    final Ehcache<String, String> ehcache = this.getEhcache(null);
+    
+    try {
+      ehcache.putIfAbsent(null, null);
+      fail();
+    } catch (NullPointerException e) {
+      // expected
+    }
+    verify(testCacheEventListener,never()).onEvent(any(CacheEvent.class));
+   
+  }
+
+
+  /**
```

Comment:
not sure it makes sense to include the cacheloaderwriter for this test, it seems that we should test hit, miss and failure (exceptions)
from the javadoc, exceptions come from null key or value (tested above) or cacheloaderwriter exceptions (CacheLoadingException and CacheWritingException)